tests/integration/os.rs

```diff
@@ -91,6 +100,62 @@ async fn return_result_cairo1_account(
     .expect("OS run failed");
 }
 
+#[rstest]
+// We need to use the multi_thread runtime to use task::block_in_place for sync -> async calls.
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn declare_cairo1_account(
+    #[future] initial_state_cairo1: StarknetTestState,
+    block_context: BlockContext,
+    max_fee: Fee,
+) {
+    let mut initial_state = initial_state_cairo1.await;
+
+    let tx_version = TransactionVersion::TWO;
+    let mut nonce_manager = NonceManager::default();
+
+    let account_contract = initial_state.cairo1_contracts.get("account_with_dummy_validate").unwrap();
+
+    // we want to declare a fresh (never-before-declared) contract, so we don't want to reuse
+    // anything from the test fixtures, and we need to do it "by hand"
+    let (_, sierra_class, casm_class) = load_cairo1_contract("empty_contract");
+    let mut ffc: FactFetchingContext<DictStorage, PoseidonHash> =
+        initial_state.cached_state.state.ffc.clone_with_different_hash();
```

Comment:
Nitpick: make an accessor method on `initial state`?

Reply:
Good idea, added in f392552.